tests/system_tests_three_routers.py

```diff
@@ -373,5 +433,189 @@ def run(self):
+
+
+    def poll_timeout ( self ) :
+        self.poll()
+
+
+    def on_start(self, event):
+        # Iff this timer expires, the test fails.
+        self.timer = event.reactor.schedule ( TIMEOUT, Timeout(self) )
+
+        # At startup, create only the receivers and the sender that checks for
+        # the route being ready.  Creation of the payload message sender
+        # has to wait until we know the address is available, to avoid a
+        # 'no route to destination' error.
+        self.recv_connection = event.container.connect ( self.route_container )
+
+        self.route_check_connection  = event.container.connect ( self.route_check_address )
+        self.route_check_receiver = event.container.create_receiver ( self.route_check_connection,
+                                                                dynamic=True
+                                                              )
+        self.route_check_sender = event.container.create_sender ( self.route_check_connection,
+                                                                  "$management"
+                                                                )
+
+
+
+    def poll ( self ):
+        # The router prepends this 'D' to the address.  If we want
+        # to look for it down at router A, we have to do likewise.
+        doctored_linkroute_prefix = 'D' + self.linkroute_prefix
+        request = self.proxy.read_address ( doctored_linkroute_prefix )
+        self.route_check_sender.send ( request )
+
+
+
+    def on_link_opened ( self, event ):
+
+        if event.receiver:
+            event.receiver.flow ( self.count )
+
+        if event.receiver == self.route_check_receiver:
+            self.proxy = RouterProxy ( self.route_check_receiver.remote_source.address )
+            self.poll()
+
+
+    def on_link_opening ( self, event ):
+        if event.receiver:
+            event.receiver.target.address = self.route_container
+            event.receiver.flow(1)
+
+
+    def on_sendable ( self, event ):
+        if event.sender == self.sender and self.sent < self.count :
+            self.sender.send ( Message ( address = self.linkroute_prefix, body = self.true_statement ) )
+            self.sent += 1
+
+
+    def on_message ( self, event ):
+        # Check that the incoming message is the one we sent.
+        if ( event.message.body == self.true_statement ) :
+            self.bail ( None )
+
+        if event.receiver == self.route_check_receiver:
+            response = self.proxy.response(event.message)
+            if response.status_code == 200 and (response.remoteCount + response.containerCount) > 0:
+                if self.route_check_timer :
+                    self.route_check_timer.cancel()
+                    self.route_check_timer = None
+                # After address appears at the far end of my router line, (router A)
+                # give more time to let it finish propagating.
+                time.sleep ( 2 )
```

Comment:
Is this sleep really necessary?  Why do you need to allow more time for propagation when you have just verified that the propagation is complete?